pkg/loader/base.go

```diff
@@ -106,12 +106,17 @@ func ValidateInputTypes(names []string) error {
 type LoadedConfigurations interface {
 	// AddConfiguration adds a configuration entry for the given path
 	AddConfiguration(path string, config IACConfiguration)
+	// ConfigurationPath checks if the given path has already been loaded as a
+	// part of another IACConfiguration, and if so, returns the path for that
+	// configuration.
+	ConfigurationPath(path string) *string
+	// AlreadyLoaded indicates whether the given path has already been loaded as
+	// part of another IACConfiguration.
+	// NOTE: consider removing this in favor of `ConfigurationPath`.
```

Comment:
I like that you've combined the implementations. My opinion is that the way that we're currently using `AlreadyLoaded`, the name of the method and that it returns a boolean make the intent of that code clearer. I'm open to being persuaded, but I'm inclined to keep this unless we change the code that uses it somehow.